core/src/main/scala/cats/instances/future.scala

```diff
@@ -8,12 +8,18 @@ import scala.concurrent.{ExecutionContext, Future}
 
 trait FutureInstances extends FutureInstances1 {
 
-  implicit def catsStdInstancesForFuture(implicit ec: ExecutionContext): MonadError[Future, Throwable] with CoflatMap[Future] =
-    new FutureCoflatMap with MonadError[Future, Throwable]{
+  implicit def catsStdInstancesForFuture(implicit ec: ExecutionContext): MonadError[Future, Throwable] with CoflatMap[Future] with MonadRec[Future] =
+    new FutureCoflatMap with MonadError[Future, Throwable] with MonadRec[Future] {
       def pure[A](x: A): Future[A] = Future.successful(x)
 
       def flatMap[A, B](fa: Future[A])(f: A => Future[B]): Future[B] = fa.flatMap(f)
 
+      final def tailRecM[B, C](b: B)(f: B => Future[(B Xor C)]): Future[C] =
```

Comment:
@johnynek I'm not finding the discussion I was thinking of at the moment, but I did try this on 2.9.3:
```scala
import scala.concurrent.{ Await, Future }, scala.concurrent.duration._
import scala.concurrent.ExecutionContext.Implicits.global

def test(i: Int)(f: Int => Future[Int]): Future[Int] = f(i).flatMap {
  case x if x > 100000 => Future.successful(x)
  case x => test(x)(f)
}

val f1 = test(0)(x => Future(x + 1))
val f2 = test(0)(x => Future(x))
```

Neither blows the stack, and `f1` succeeds as expected.